src/core/renderer.js

```diff
@@ -141,58 +140,29 @@ class Renderer {
         this._zoom = zoom;
     }
 
-
-    /**
-     * Removes a dataframe for the renderer. Freeing its resources.
-     * @param {*} tile
-     */
-    removeDataframe(dataframe) {
-        this.dataframes = this.dataframes.filter(t => t !== dataframe);
-    }
-
-    /**
-     * @description Adds a new dataframe to the renderer.
-     *
-     * Performance-intensive. The required allocation and copy of resources will happen synchronously.
-     * To achieve good performance, avoid multiple calls within the same event, particularly with large dataframes.
-     * @param {Dataframe} dataframe
-     * @returns {BoundDataframe}
-     */
-    addDataframe(dataframe) {
-        dataframe.bind(this);
-        this.dataframes.push(dataframe);
-    }
-
     getAspect() {
         if (this.gl) {
             return this.gl.canvas.clientWidth / this.gl.canvas.clientHeight;
         }
         return 1;
     }
 
-
-    getStyledTiles() {
-        return this.dataframes.filter(tile => tile.style && tile.visible && tile.style.colorShader);
-    }
-
-
-    _computeDrawMetadata() {
+    _computeDrawMetadata(layer) {
+        const tiles = layer.dataframes;
```

Comment:
I think we should rename tiles to `dataframes`

Reply:
Yes, but that's outside this PR scope